test/Network/MQTT/EncoderSpec.hs

```diff
@@ -64,5 +59,15 @@ instance Monad m => Serial m ClientIdentifier
 instance Monad m => Serial m PacketIdentifier
 instance Monad m => Serial m UserName
 instance Monad m => Serial m Password
-instance Monad m => Serial m Topic
-instance Monad m => Serial m TopicFilter
+instance Monad m => Serial m Topic where
+  series = do
+    s <- Topic <$> series
+    if unTopic s == ""
+      then return (Topic $ T.pack "My topic")
```

Comment:
I couldn't come up with the better approach since in order to generate only non-empty text we need to change Text instance which is inside different package.